frontend/pages/admin/IntegrationsPage/cards/MdmSettings/MacOSMdmPage/MacOSMdmPage.tsx

```diff
@@ -0,0 +1,171 @@
+        install software on your macOS hosts.
+      </p>
+      <ol className={`${baseClass}__setup-instructions-list`}>
+        <li>
+          <p>
+            1. Request a certificate signing request (CSR) and key for Apple
+            Push Notification Service (APNs) and a certificate and key for
+            Simple Certificate Enrollment Protocol (SCEP).
+          </p>
+          <Button
+            className={`${baseClass}__request-button`}
+            onClick={onClickRequest}
+            variant="brand"
+          >
+            Request
+          </Button>
+        </li>
+        <li>
+          <p>2. Go to your email to download your CSR.</p>
+        </li>
+        <li>
+          <p>
+            3.{" "}
+            <CustomLink
+              url="https://identity.apple.com/pushcert/"
+              text="Sign in to Apple Push Certificates Portal"
+              newTab
+            />
+            <br />
+            If you don&apos;t have an Apple ID, select <b>Create yours now</b>.
+          </p>
+        </li>
+        <li>
+          <p>
+            4. In Apple Push Certificates Portal, select{" "}
+            <b>Create a Certificate</b>, upload your CSR, and download your APNs
+            certificate.
+          </p>
+        </li>
+        <li>
+          <p>
+            5. Deploy Fleet with <b>mdm</b> configuration.{" "}
+            <CustomLink
+              url="https://fleetdm.com/docs/deploying/configuration#mobile-device-management-mdm"
+              text="See how"
+              newTab
+            />
+          </p>
+        </li>
+      </ol>
+    </div>
+  );
+};
+
+interface IApplePushCertificatePortalSetupInfoProps {
+  appleAPNInfo: IMdmApple;
+}
+
+const ApplePushCertificatePortalSetupInfo = ({
+  appleAPNInfo,
+}: IApplePushCertificatePortalSetupInfoProps) => {
+  return (
+    <dl className={`${baseClass}__page-content ${baseClass}__apc-info`}>
+      <div>
+        <dt>Common name (CN)</dt>
+        <dd>{appleAPNInfo.common_name}</dd>
+      </div>
+      <div>
+        <dt>Serial number</dt>
+        <dd>{appleAPNInfo.serial_number}</dd>
+      </div>
+      <div>
+        <dt>Issuer</dt>
+        <dd>{appleAPNInfo.issuer}</dd>
+      </div>
+      <div>
+        <dt>Renew date</dt>
+        <dd>{readableDate(appleAPNInfo.renew_date)}</dd>
+      </div>
+    </dl>
+  );
+};
+
+const MacOSMdmPage = () => {
+  const [showRequestCSRModal, setShowRequestCSRModal] = useState(false);
+
+  // Currently the status of this API call is what determines various UI states on
+  // this page. Because of this we will not render any of this components UI until this API
+  // call has completed.
+  const {
+    data: appleAPNInfo,
+    isLoading: isLoadingMdmApple,
+    error: errorMdmApple,
+  } = useQuery<IMdmApple, AxiosError, IMdmApple>(["appleAPNInfo"], () =>
+    mdmAppleAPI.getAppleAPNInfo()
+  );
+
+  const toggleRequestCSRModal = () => {
+    setShowRequestCSRModal(!showRequestCSRModal);
```

Comment:
Nit: I like the updater pattern because it removes the need to worry about potential state dependencies for these sorts of toggles: `setShowRequestCSRModal((prevState) => !prevState)`;